pkg/registry/generic/registry/store.go

```diff
@@ -479,6 +479,11 @@ func markAsDeleting(obj runtime.Object) (err error) {
 		return kerr
 	}
 	now := unversioned.NewTime(time.Now())
+  // This handles Generation bump caused by the finalizer-driven cascading deletion,
```

Comment:
I think the following description is more accurate: "This handles Generation bump for resources that don't support graceful deletion. For resources that support graceful deletion is handle in pkg/api/rest/delete.go". WDYT?